src/library_syscall.js

```diff
@@ -1037,26 +1037,21 @@ var SyscallsLibrary = {
       var id;
       var type;
       var name = stream.getdents[idx];
-      if (name[0] === '.') { // XXX name? see https://github.com/emscripten-core/emscripten/issues/7487
```

Comment:
I think perhaps this code was supposed to be `if (name == '.' or name == '..')`

Reply:
That could be. But would the `id` variable be `1` in both those cases?